providers/pool.rb

```diff
@@ -120,120 +120,152 @@ def site_identifier
+      cmd << " /processModel.maxProcesses:#{@new_resource.max_proc}" if @new_resource.max_proc && is_new_max_processes
+      cmd << " /queueLength:#{@new_resource.queue_length}" if @new_resource.queue_length && is_new_queue_length
       Chef::Log.debug(cmd)
       shell_out!(cmd)
-      @new_resource.updated_by_last_action(true)
-    end
-    if @new_resource.runtime_version && is_new_managed_runtime_version
-      was_updated = true
-      cmd = "#{appcmd(node)} set apppool \"/apppool.name:#{@new_resource.pool_name}\""
-      cmd << " /managedRuntimeVersion:v#{@new_resource.runtime_version}"
-      Chef::Log.debug(cmd) if @new_resource.runtime_version
-      shell_out!(cmd)
-      @new_resource.updated_by_last_action(true)
-    end
-    if @new_resource.worker_idle_timeout && is_new_idle_timeout
-      was_updated = true
-      cmd = "#{appcmd(node)} set config /section:applicationPools"
-      cmd << " \"/[name='#{@new_resource.pool_name}'].processModel.idleTimeout:#{@new_resource.worker_idle_timeout}\""
-      Chef::Log.debug(cmd)
-      shell_out!(cmd)
-      @new_resource.updated_by_last_action(true)
     end
+
+    # Application Pool Config
+    # processModel items
+    configure_application_pool(@new_resource.worker_idle_timeout && is_new_idle_timeout, "processModel.idleTimeout:#{@new_resource.worker_idle_timeout}")
+    configure_application_pool(is_new_load_user_profile, "processModel.loadUserProfile:#{@new_resource.load_user_profile}")
+    configure_application_pool(is_new_set_profile_environment, "processModel.setProfileEnvironment:#{@new_resource.set_profile_environment}")
+    configure_application_pool(is_new_logon_type, "processModel.logonType:#{@new_resource.logon_type}")
+    configure_application_pool(is_new_manual_group_membership, "processModel.manualGroupMembership:#{@new_resource.manual_group_membership}")
+    configure_application_pool(is_new_idle_timeout, "processModel.idleTimeout:#{@new_resource.idle_timeout}")
+    configure_application_pool(is_new_shutdown_time_limit, "processModel.shutdownTimeLimit:#{@new_resource.shutdown_time_limit}")
+    configure_application_pool(is_new_startup_time_limit, "processModel.startupTimeLimit:#{@new_resource.startup_time_limit}")
+    configure_application_pool(is_new_pinging_enabled, "processModel.pingingEnabled:#{@new_resource.pinging_enabled}")
+    configure_application_pool(is_new_ping_interval, "processModel.pingInterval:#{@new_resource.pinging_interval}")
+    configure_application_pool(is_new_ping_response_time, "processModel.pingResponseTime:#{@new_resource.pinging_response_time}")
+    
+    # recycling items
+    configure_application_pool(is_new_log_event_on_recycle, "recycling.logEventOnRecycle:PrivateMemory,Memory,Schedule,Requests,Time,ConfigChange,OnDemand,IsapiUnhealthy")
+    configure_application_pool(@new_resource.private_mem && is_new_private_memory, "recycling.periodicRestart.privateMemory:#{@new_resource.private_mem}")
```

Comment:
if they have a default they don't need the variable check, if they do have default then it is just is_new